syntax/value.php

```diff
@@ -0,0 +1,136 @@
+         */
+        return 155;
+    }
+
+    /**
+     * Connect lookup pattern to lexer.
+     *
+     * @param string $mode Parser mode
+     */
+    public function connectTo($mode) {
+        /*
+         * Full syntax:
+         * ---- struct value ----
+         * <config>
+         * ----
+         */
+        $this->Lexer->addSpecialPattern('----+ *struct value *-+\n.*?\n----+', $mode, 'plugin_struct_value');
+
+        /*
+         * Inline syntax (page ID will default to $ID$ if not supplied):
+         * {{$<schema>.<field>}}
+         * {{$<pageid>.<schema>.<field>}}
+         * Any <component> can be surrounded by quotes - see http://php.net/manual/function.str-getcsv.php
+         */
+        $this->Lexer->addSpecialPattern('\{\{\$[^}]+\}\}', $mode, 'plugin_struct_value');
+    }
+
+    /**
+     * Handle matches of the struct syntax
+     *
+     * @param string $match The match of the syntax
+     * @param int $state The state of the handler
+     * @param int $pos The position in the document
+     * @param Doku_Handler $handler The handler
+     * @return array Data for the renderer
+     */
+    public function handle($match, $state, $pos, Doku_Handler $handler) {
+        global $conf;
+
+        try {
+            // Choose which parser to use
+            if (substr($match, 0, 3) == '{{$') {
+                // strip {{$ and }} markers
+                $inline = substr($match, 3, -2);
+
+                // Parse inline syntax
+                $parser = new InlineConfigParser($inline);
+            } else {
+                $lines = explode("\n", $match);
+    
+                // Strip off delimiter lines
+                array_shift($lines);
+                array_pop($lines);
+
+                // Parse full syntax
+                $parser = new ConfigParser($lines);
+            }
+
+            $config = $parser->getConfig();
+            return $config;
+        } catch(StructException $e) {
+            msg($e->getMessage(), -1, $e->getLine(), $e->getFile());
+            if($conf['allowdebug']) msg('<pre>' . hsc($e->getTraceAsString()) . '</pre>', -1);
+            return null;
+        }
+    }
+
+    /**
+     * Render xhtml output or metadata
+     *
+     * @param string $mode Renderer mode (supported modes: xhtml)
+     * @param Doku_Renderer $renderer The renderer
+     * @param array $data The data from the handler() function
+     * @return bool If rendering was successful.
+     */
+    public function render($mode, Doku_Renderer $renderer, $data) {
+        if(!$data) return false;
+        global $INFO;
+        global $conf;
+
+        // Get configuration
+        $show_not_found = $this->getConf('show_not_found');
+
+        try {
+            $search = new SearchConfig($data);
+
+            /** @var AggregationValue $value */
+            $value = new AggregationValue($INFO['id'], $mode, $renderer, $search);
+            $value->render($show_not_found);
```

Comment:
the above does not need to happen during metadata rendering.

Reply:
Is this not where the value would be put into the text of the abstract when rendering metadata? This is quite a complex part of the DokuWiki code and I'm not at all sure I understand it!

Reply:
Hmm, I think you might be right. I forgot about that. Maybe give it a quick try by putting the value right at the top of the page and see what ends up in the abstract